### ISSUE TITLE
Allow limiting fps to vsync

### ISSUE THREAD

**Issue by AaronVanGeffen** [OpenLoco/OpenLoco#1945]
Currently, OpenLoco either limits the fps to 40fps, or does not limit things at all. It seems more sensible to limit the game to vsync by default, while still allowing the 'vanilla', non-interpolated 40fps experience, and 'unlimited' mode for benchmarks.

**Comment by voidxor** [comment]
@AaronVanGeffen I see that this was implemented last month, and thus should be complete in version 26.01. However, whenever I choose "Vertical sync" in the options, it stays that way for that session only. My monitor is set to 60 Hz. Upon closing the game and reopening, it changes to "Unrestricted" frame rate, so my processor has to work three times as hard to render something that my monitor cannot display.

It's as if selecting "Vertical sync" in the options actually gets saved to disk as "Unrestricted".

Full disclosure: I've had RCT since 1999, yet have only discovered OpenRCT2, Locomotion, and OpenLoco in recent months. So I'll confess to being a Locomotion novice. I first downloaded OpenLoco starting with the release of 26.01 over the weekend. I love that you guys are keeping Chris Sawyer's masterpieces up to date!

I likewise don't have much GitHub experience, so forgive me if I should've created a new issue, rather than comment on this one. It might need to be reopened, though.

**Comment by AaronVanGeffen** [comment]
Yes, there is a bug saving the setting in the 26.01 release. We fixed it the other day (#3626), so it will be fixed in the next release.

In the mean time, you could edit the openloco.yml config file manually to force vsync on.

**Comment by voidxor** [comment]
Adding "vsync: true" to the YML file beneath the "display:" heading worked, thanks.

I see in both Config.cpp and openloco.yml that the _mode_, _window_resolution_, _fullscreen_resolution_, and _vsync_ options are nested under the "display:" heading, while _scale_factor_, _showFPS_, and _uncapFPS_ are not. Should they be grouped with the other display settings?

**Comment by AaronVanGeffen** [comment]
Thanks for confirming. Yes, many of the settings are in the 'general' struct when they could. be grouped in smaller chunks. I'd prefer to keep them as-is, for now, mostly for backwards compatibility reasons. There are (vague) plans to move the config serialisation to a different file format, and so I'd prefer to tackle it when we do switch.
